modules/oke/nodepools.tf

```diff
@@ -53,6 +54,7 @@ resource "oci_containerengine_node_pool" "nodepools" {
     size = max(0, lookup(each.value, "node_pool_size", 0))
 
     freeform_tags = var.freeform_tags["persistent_volume"]
```

Comment:
I think using the persistence_volume tag here was a mistake. It shouldn't have been used here. Please remove the equivalent for `defined_tags` as well.

Reply:
It could be useful to specify tags for single nodes (well, I am actually using this feature), so I just added the key "node" to the map for both freeform and defined tags

Reply:
I see we already have 1 for `node_pool`. What will be the difference between the 2?

Reply:
You can apply node_pool tags to their nodes but an user could want to add specific tags to node instances, for example to perform queries on instances using the search engine and for reporting.
Also, I think that this also reflects the behavior of OCI web interface, through which you can specify both node pool tags and node tags.

Reply:
I'm not sure I follow. Can you please elaborate a bit more on your use case or what you're trying to achieve with the node tag? 

I agree with the common nodepool tag but I don't quite follow the node one.

Reply:
I am simply trying to reproduce the behavior of Oracle Cloud Console, that allows the user to specify tags for nodes of a particular node pool but not for the pool itself. 

For my specific use case, I can just apply the node pool tags to the single nodes, because the most important requirement is to assign a specific tag to resources of type _instance_.
Let me know if I should delete the key `node` and use the `node_pool` instead.

Reply:
I've done another round of test and I now understand more where you're going with the node tags and you're in the right direction.

So, similar to nodepool tags, we need to have common and tags for the nodes that are specific to the nodepool. So, https://github.com/valireds/terraform-oci-oke/blob/issue-560/modules/oke/nodepools.tf#L56 and https://github.com/valireds/terraform-oci-oke/blob/issue-560/modules/oke/nodepools.tf#L57 should read similar as https://github.com/valireds/terraform-oci-oke/blob/issue-560/modules/oke/nodepools.tf#L13 and https://github.com/valireds/terraform-oci-oke/blob/issue-560/modules/oke/nodepools.tf#L14 respectively except that you'll be extracting the node key instead of node_pool.

Reply:
I should add: well done for spotting that the nodes needed tags on their own. I initially thought setting tags for the nodepools would be sufficient.

Reply:
This should be fixed in the latest version.